### PR TITLE
Allow empty Digest configuration for DELETE operation

### PR DESCRIPTION
**Pull request by gcgirish**
This PR supports empty Digest config for DELETE operation. For DELETE operation all that is needed is the DigestId.

### CODE REVIEW

**Review by pierventre — DISMISSED** [review]
do we need to update PI ?

**Comment by gcgirish** [comment]
> do we need to update PI ?

This is my first commit to Stratum project. What is PI?

**Comment by pierventre** [comment]
> > do we need to update PI ?
> 
> This is my first commit to Stratum project. What is PI?

hal/pi for bmv2 but I am not fully sure; perhaps a question for @pudelkoM

**Comment by pudelkoM** [comment]
> do we need to update PI ?

depends on if PI already handles deletes correctly. Worth checking, but what we observed right now is BFRT only.

**Comment by gcgirish** [comment]
> > do we need to update PI ?
> 
> depends on if PI already handles deletes correctly. Worth checking, but what we observed right now is BFRT only.

I cant find reference to Digest handling for BMV2 in stratum. Yes, we have seen this issue for BFRT only.

**Comment by pierventre** [comment]
> > > do we need to update PI ?
> > 
> > 
> > depends on if PI already handles deletes correctly. Worth checking, but what we observed right now is BFRT only.
> 
> I cant find reference to Digest handling for BMV2 in stratum. Yes, we have seen this issue for BFRT only.

I am confused - dont we support digests on stratum-bmv2 ? if we supprt, are we not doing DELETE in the tests ? Or is it a lucky case because the digest config is gone every time we do sw tests ?

File: stratum/hal/lib/barefoot/bfrt_table_manager_test.cc [hunk 1]
```diff
@@ -578,11 +597,6 @@ TEST_F(BfrtTableManagerTest, DeleteDigestEntrySuccess) {
 
   const std::string kDigestEntryText = R"pb(
     digest_id: 401732455
-    config {
```
Comment (pierventre) [inline stratum/hal/lib/barefoot/bfrt_table_manager_test.cc:581]: shall we maintain this test and verify that brft_table_manager does not care? It seems to me another test we get for free.

**Review by pierventre — DISMISSED** [review]

**Comment by pudelkoM** [comment]
> > > > do we need to update PI ?
> > > 
> > > 
> > > depends on if PI already handles deletes correctly. Worth checking, but what we observed right now is BFRT only.
> > 
> > 
> > I cant find reference to Digest handling for BMV2 in stratum. Yes, we have seen this issue for BFRT only.
> 
> I am confused - dont we support digests on stratum-bmv2 ? if we supprt, are we not doing DELETE in the tests ? Or is it a lucky case because the digest config is gone every time we do sw tests ?

We do, but Stratum-bmv2 to PI is nearly completely pass-through from the Stratum view. We don't even look into the P4 entities before handing them over. If you're referring to the unit tests modified here, those are for bfrt only.

**Comment by pierventre** [comment]
> > > > > do we need to update PI ?
> > > > 
> > > > 
> > > > depends on if PI already handles deletes correctly. Worth checking, but what we observed right now is BFRT only.
> > > 
> > > 
> > > I cant find reference to Digest handling for BMV2 in stratum. Yes, we have seen this issue for BFRT only.
> > 
> > 
> > I am confused - dont we support digests on stratum-bmv2 ? if we supprt, are we not doing DELETE in the tests ? Or is it a lucky case because the digest config is gone every time we do sw tests ?
> 
> We do, but Stratum-bmv2 to PI is nearly completely pass-through from the Stratum view. We don't even look into the P4 entities before handing them over. If you're referring to the unit tests modified here, those are for bfrt only.

No, I am referring to the recovery-nat tests that are failing due to this and I am wondering why the same tests on software are not

**Comment by gcgirish** [comment]
> > > > > > do we need to update PI ?
> > > > > 
> > > > > 
> > > > > depends on if PI already handles deletes correctly. Worth checking, but what we observed right now is BFRT only.
> > > > 
> > > > 
> > > > I cant find reference to Digest handling for BMV2 in stratum. Yes, we have seen this issue for BFRT only.
> > > 
> > > 
> > > I am confused - dont we support digests on stratum-bmv2 ? if we supprt, are we not doing DELETE in the tests ? Or is it a lucky case because the digest config is gone every time we do sw tests ?
> > 
> > 
> > We do, but Stratum-bmv2 to PI is nearly completely pass-through from the Stratum view. We don't even look into the P4 entities before handing them over. If you're referring to the unit tests modified here, those are for bfrt only.
> 
> No, I am referring to the recovery-nat tests that are failing due to this and I am wondering why the same tests on software are not

I think bmv2 does not validate this and stratum is transparent to any requests going to bmv2.

**Review by pierventre — APPROVED** [review]

**Pull request merged**